functions/Find-DbaAgentJob.ps1

```diff
@@ -192,8 +196,35 @@ Queries CMS server to return all SQL instances in the Production folder and then
                                     {
                                         $DynString += '-and $Category -contains $_.Category '
                                     }
-
-
+                            }
+                        IF ($Owner)
+                            {
+                                Write-Verbose "Owner tagged in combine filter"
+                                IF ($Owner -match "-")
+                                    {
+                                        $OwnerMatch = $Owner -replace "-", ""
+                                        Write-Verbose "Checking for jobs that NOT owned by: $Owner"
+                                        IF ($filter -eq 0)
+                                            {
+                                                $DynString += ' | Where-Object { $OwnerMatch -notcontains $_.OwnerLoginName '
+                                            }
+                                        ELSEIF ($filter -eq 1)
+                                            {
+                                                $DynString += '-and $OwnerMatch -notcontains $_.OwnerLoginName '
+                                            }
+                                    }
+                                ELSE
+                                    {
+                                        Write-Verbose "Checking for jobs that are owned by: $Owner"
+                                        IF ($filter -eq 0)
+                                            {
+                                                $DynString += ' | Where-Object { $Owner -contains $_.OwnerLoginName '
+                                            }
+                                        ELSEIF ($filter -eq 1)
+                                            {
+                                                $DynString += '-and $Owner -contains $_.OwnerLoginName '
+                                            }
+                                    }
                             }
 
                         IF (!($DynString -eq '$output = $server.JobServer.jobs'))
```

Comment:
this means "when we built a string that's more complex than return all jobs", right?
IF ( $DynString -ne '$output = $server.JobServer.jobs') is a bit more direct